### ISSUE TITLE
Blazor WASM Debugging in VS launches Edge with a separate user data directory

### ISSUE THREAD

**Issue by anurse** [dotnet/aspnetcore#20915]
@rynowak told me to just file something here to start with :). I'm sure this lives somewhere else, like in VS.

If I launch a Blazor WASM app in VS with Ctrl-F5, I get a new tab opened in my existing Edge browser window and everything is nice. My extensions are present, and I'm properly signed in to Edge, you can see my profile picture in the top-right:

![The top-right of a Microsoft Edge browser window showing my profile picture](https://user-images.githubusercontent.com/7574/79501567-1c75b000-7fe3-11ea-8a65-105f74836af2.png)

Cool cool cool cool.

Now, I launch with F5 to use the swanky new debugging features. Instead of a new tab, I get a new window. That's fine, I don't mind that. However, it's not signed in with my profile, so I don't get my extensions or existing cookies. This was annoying because I'm building an app that signs in to GitHub, and I didn't have my Github cookie or my password manager in order to log in again. Since I'm not signed in, my profile picture isn't in the top right:

![The top-right of a Microsoft Edge browser window showing no profile picture](https://user-images.githubusercontent.com/7574/79501774-75454880-7fe3-11ea-8891-1b9088a9fefd.png)

I noticed that VS launched `msedge` with a `--user-data-dir=C:\Users\anurse\AppData\Local\Microsoft\VisualStudio\16.0_019f56a0\WebTools\EdgeUserData_E906737D_E10E9004-4C41-47FB-B7F0-6B3B57970109` parameter which seems to be why it's not picking up my normal profiles. This seems odd to me :(.

**Comment by mkArtakMSFT** [comment]
Thanks @anurse.
I will follow up with the JS Debugger team regarding this.

**Comment by SteveSandersonMS** [comment]
As I understand it, it’s necessary to use a separate profile. Chromium browsers only support a single instance per profile directory, so if you already have a browser open then ask it to open a new one with different command line options (eg to open the debugging port) it will ignore the different options and continue using the existing instance. 

The only way we can be sure to enable debugging is either:

- Force the user to close all existing browser instances (extremely hostile)
- Or, launch on a separate profile

**Comment by digeff** [comment]
What @SteveSandersonMS said is correct. If you want to use your existing user data folder with your extensions and preference you'll have to close all existing browser instances before launching it from Visual Studio.

To configure Visual Studio to launch the browser with your existing user data folder you'll need to:
1. Go to the Browse with... option:
![image](https://user-images.githubusercontent.com/14098140/79513042-bfd0c000-7ff7-11ea-90f5-2edcc4bd3891.png)
2. Add... a new browser:
![image](https://user-images.githubusercontent.com/14098140/79513032-ba737580-7ff7-11ea-9741-f1973b468d9f.png)
3. Configure the path to your browser, and use a custom ---user-data-dir argument pointing to your user data folder
![image](https://user-images.githubusercontent.com/14098140/79513211-1d650c80-7ff8-11ea-81c5-5ab6e204a22d.png)
In my PC they are:
C:\Program Files (x86)\Microsoft\Edge Beta\Application\msedge.exe
--user-data-dir="C:\Users\<username>\AppData\local\Microsoft\Edge Beta\User Data"
4. Ok to add it, and then optionally set it as the Default browser if you choose to.
5. Browse with your user profile

**Comment by anurse** [comment]
Ugh. That feels like a really unpleasant user experience. I get the issues, but from a user experience side it feels really unsatisfactory.

I’ve made it work (by logging in to the “alternate” profile) so I don’t really need to set up the new browser launch profile. I’m not sure most users would understand what’s going on here and why they don’t have any of their cookies and extensions.

**Comment by digeff** [comment]
The 2 main restrictions that we have are:
1: If Chrome was launched without a debugging port (this is the normal launch), you cannot add one later, so we won't be able to debug it.
2. If you launch Chrome with a user profile that is already being used by another instance of Chrome, the existing instance is reused, so if Chrome is already running for that user profile, and it doesn't have a debugging port, you cannot add one unless you stop all instances of Chrome for that user profile first

The only alternative I can think of to avoid that experience would be for the user to chose a fixed port for Chrome, and modify all Chrome shortcuts to always open debugging with that fixed port, and then modify VS to also use that port too.

@anurse If you have any suggestions as to how we can improve the experience, we'd love to hear them :)

**Comment by anurse** [comment]
Yeah, it certainly does seem like we’re stuck with what we’ve got :(. Maybe some way of notifying the user would help? And/or docs clarification?

And of course we could see about filing a chromium bug but I’m not super hopeful about that being resolved any time soon (if at all :)).

**Comment by mkArtakMSFT** [comment]
Let's wait and see how community feels about this. Moving this to the backlog (as a potential docs improvement).

**Comment by arthurzaczek** [comment]
I followed these instructions and it worked for me. It would be really helpful, if this is documented in the debugging section.

I would suggest to add another browser profile called "Microsoft Edge with Dev-Profile" or something.

**Comment by TanvirArjel** [comment]
**Blazor WASM debugging should open in a new tab of the already opened window of the browser like ASP.NET Core MVC, Razor Pages app.**

**Comment by TanayParikh** [comment]
@guardrex could we add some details to the debugging docs about how a new Edge/Chrome window with a separate "dev-profile" will be opened when debugging? This is pretty low priority though 😄
